elastic/apm-server
# Remove golint, just use staticcheck

## Motivation/summary

golint is deprecated and frozen (https://github.com/golang/go/issues/38968). Remove its use, along with reviewdog, and turn on all (but one) staticcheck checks to compensate. I've fixed a few small issues that staticcheck found as a result of this.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

N/A

(If you really want you can introduce a lint issue and check that it gets picked up, e.g. change a type's comment such that it doesn't begin with the type name.)

## Related issues

https://github.com/elastic/apm-server/issues/6858
